packages/astro/src/integration/snippets.ts

```diff
@@ -0,0 +1,47 @@
+import type { SentryOptions } from './types';
+
+/**
+ * Creates a snippet that imports a Sentry.init file.
+ */
+export function buildSdkInitFileImportSnippet(filePath: string): string {
+  return `import "${filePath}";`;
+}
+
+/**
+ * Creates a snippet that initializes Sentry on the client by choosing
+ * default options.
+ */
+export function buildClientSnippet(options: SentryOptions): string {
+  return `import * as Sentry from "@sentry/astro";
+
+Sentry.init({
+  dsn: ${options.dsn ? JSON.stringify(options.dsn) : 'import.meta.env.PUBLIC_SENTRY_DSN'},
+  debug: ${options.debug ? true : false},
+  environment: ${options.environment ? JSON.stringify(options.environment) : 'import.meta.env.PUBLIC_VERCEL_ENV'},
+  release: ${options.release ? JSON.stringify(options.release) : 'import.meta.env.PUBLIC_VERCEL_GIT_COMMIT_SHA'},
+  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
+  tracesSampleRate: ${options.tracesSampleRate ?? 1.0},
+  replaysSessionSampleRate: ${options.replaysSessionSampleRate ?? 0.1},
+  replaysOnErrorSampleRate: ${options.replaysOnErrorSampleRate ?? 1.0},${
+    options.sampleRate ? `\n  sampleRate: ${options.sampleRate},` : ''
+  }
+});`;
+}
+
+/**
+ * Creates a snippet that initializes Sentry on the server by choosing
+ * default options.
+ */
+export function buildServerSnippet(options: SentryOptions): string {
+  return `import * as Sentry from "@sentry/astro";
+
+Sentry.init({
+  dsn: ${options.dsn ? JSON.stringify(options.dsn) : 'import.meta.env.PUBLIC_SENTRY_DSN'},
```

Comment:
l: Feel free to ignore, but maybe we can extract the common config stuff out into a helper function, so we don't need to duplicate this & have an easier time keeping this in sync. So e.g.:

```js
// client
return `import * as Sentry from '@sentry/astro';

Sentry.init({
  ${getCommonInitOptions(options)}
  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
  // and other client-specific config
});
```

Reply:
Good idea!